### PR TITLE
Initial Mapping-IO support

### LINKED ISSUE #402: Migrate to use the https://github.com/FabricMC/mapping-io libary
The https://github.com/FabricMC/mapping-io can replace the current mapping IO in engima.

### PR DESCRIPTION
**Pull request by NebelNidas**
This PR adds full Mapping-IO import and export support. It doesn't replace Enigma's own (legacy) import/export code (yet), so you can choose which library to use, in case there are still bugs in the new solution. The CLI defaults to the MIO implementations, this can be changed via the `enigma.use_mappingio` System property.

Closes #402 and supersedes #465.

### CODE REVIEW

**Comment by modmuss50** [comment]
At a glace this looks like a great start.

Most of the time we dont use the GUI to select the mappings, in yarn its passed as run argument, or the commands to augment the mappings are directly invoked. It might be nice to have a system property? to make it the default option so we could try it out easily in yarn.

File: gradlew.bat [hunk 1]
```diff
@@ -1,92 +1,92 @@
...
-set EXIT_CODE=%ERRORLEVEL%
-if %EXIT_CODE% equ 0 set EXIT_CODE=1
-if not ""=="%GRADLE_EXIT_CONSOLE%" exit %EXIT_CODE%
-exit /b %EXIT_CODE%
-
-:mainEnd
-if "%OS%"=="Windows_NT" endlocal
-
-:omega
+@rem
```
Comment (modmuss50) [inline gradlew.bat:1]: Dont change the line endings of this file. If there is something wrong with it fix it in another PR.

Comment (NebelNidas) [reply gradlew.bat:1]: This seems to happen automatically, probably due to the gitattributes configuration

Comment (NebelNidas) [reply gradlew.bat]: I wasn't able to fix it, Git just immediately re-applied the change making it impossible for me to rebase. Seemingly only happens on Windows. Fortunately the upstream merge overwrote the it

Comment (modmuss50) [reply gradlew.bat]: Thanks, should now be correct.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**